src/shogun/kernel/CombinedKernel.h

```diff
@@ -124,7 +124,14 @@ class CCombinedKernel : public CKernel
 		 */
 		inline CKernel* get_kernel(int32_t idx)
 		{
-			return (CKernel*) kernel_array->get_element(idx);
+			if (idx < get_num_kernels())
+			{
+				return (CKernel*)kernel_array->get_element(idx);
+			}
+			else
+			{
+				return 0;
```

Comment:
I agree with you -- however the implementation of combined kernel seems to expect 0 being returned if its out of bounds, hence why i went with that.